### PR TITLE
Improve code coverage for pinot-common

### PR DESCRIPTION
**Pull request by bradengroom**
## Description
This adds some tests for uncovered lines in `pinot/common/function` and updates `.codecov.yml` to ignore all generated files in `pinot/common/request`. This works towards [the goal of getting coverage for pinot-common up to 75%](https://github.com/apache/incubator-pinot/issues/5695).

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/function/AggregationFunctionType.java [hunk 1]
```diff
@@ -75,15 +75,6 @@ public String getName() {
...
   }
 
-  public boolean isOfType(AggregationFunctionType... aggregationFunctionTypes) {
-    for (AggregationFunctionType aggregationFunctionType : aggregationFunctionTypes) {
-      if (this == aggregationFunctionType) {
-        return true;
-      }
-    }
-    return false;
-  }
```
Comment (bradengroom) [inline pinot-common/src/main/java/org/apache/pinot/common/function/AggregationFunctionType.java:85]: This appears to be untested, but also unused. Is there a reason to keep this?

**Comment by bradengroom** [comment]
This coverage report doesn’t seem correct. It reports drops in coverage for files I haven’t touched?

File: pinot-common/src/main/java/org/apache/pinot/common/function/AggregationFunctionType.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-common/src/main/java/org/apache/pinot/common/function/AggregationFunctionType.java:85]: We can remove it

**Review by Jackie-Jiang — APPROVED** [review]
LGTM, thanks.
Not sure why the coverage report is not stable

**Pull request merged**